eos_data_distribution/avahi_monitor.py

```diff
@@ -184,13 +187,22 @@ def service_removed_cb(self, sda, interface, protocol, name, type, domain, flags
             self.process_route_changes()
 
     def add_nexthop(self, faceUri):
+        if len(self._peers) > MAX_PEERS:
```

Comment:
If the code ends up calling `add_nexthop()` 6 times (i.e. there are 6 valid next hops in the network), then `remove_nexthop()` for one of them once (for example, because that host disappears), we will end up with 4 peers registered in the RIB, not 5.

Taken to the limit, it would be possible for this machine to end up with zero hops in its RIB, even if several valid hops were present on the network, just because the first 5 hops to be added here were also the first (and only) 5 to be removed.